paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -220,20 +225,25 @@ internal class PaymentOptionsViewModel @Inject constructor(
         clearErrorMessages()
 
         selection.value?.let { paymentSelection ->
-            // TODO(michelleb-stripe): Should the payment selection in the event be the saved or new item?
             eventReporter.onSelectPaymentOption(paymentSelection)
             val linkState = args.state.paymentMethodMetadata.linkState
-            if (linkState != null && shouldShowLinkVerification(paymentSelection, linkState)) {
+            val linkAccount = linkAccountHolder.linkAccount.value
+            val shouldShowLinkConfiguration = linkState != null && shouldShowLinkVerification(
+                paymentSelection = paymentSelection,
+                linkState = linkState,
+                linkAccount = linkAccount
+            )
+            if (shouldShowLinkConfiguration) {
                 linkPaymentLauncher.present(
                     configuration = linkState.configuration,
                     launchMode = LinkLaunchMode.PaymentMethodSelection(selectedPayment = null),
-                    linkAccount = null,
+                    linkAccount = linkAccount,
```

Comment:
populates the current LinkAccount info to `FlowController` even if the options screen is closed via cancellation.